Mathlib/Data/Nat/Bitwise.lean

```diff
@@ -338,6 +347,25 @@ theorem bitwise_comm {f : Bool → Bool → Bool} (hf : ∀ b b', f b b' = f b'
     _ = swap (bitwise f) := bitwise_swap
 #align nat.bitwise_comm Nat.bitwise_comm
 
+/-- If `x` and `y` fit within `n` bits, then the result of any (well-behaved) bitwise operation on
+    `x` and `y` also fits within `n` bits -/
+theorem bitwise_lt {f x y n} (hx : x < 2 ^ n) (hy: y < 2 ^ n) (h: f false false = false) :
+    bitwise f x y < 2 ^ n := by
```

Comment:
Is `h` needed here?

Reply:
Good catch, I think the result should still be true even without `h`.
That said, the current proof does rely on `h`.

Reply:
Fixed. I think the answer is always `binaryRec'`

Reply:
Thank you @eric-wieser. This is much appreciated.